std/algorithm/comparison.d

```diff
@@ -1565,3 +1571,385 @@ unittest
                 2, "two",
                 ));
 }
+
+/**
+Returns $(D true) if both ranges have the same length. This function is optimized
+to always take advantage of the $(D length) member of either range if it exists.
+
+If both ranges have a length member: $(BIGOH 1)
+If r1 doesn't have a length member: $(BIGOH r1.length)
+If r2 doesn't have a length member: $(BIGOH r2.length)
+If both ranges don't have a length member: $(BIGOH 2 * r1.length)
```

Comment:
Nice addition!

For complexity, If both ranges have a length member, it's O(1). Otherwise, it's always O(min(r1.length, r2.length)). For big-O notation, you never include constants (e.g. 2 * r1.length).